bare_metal_billing/models.py

```diff
@@ -0,0 +1,63 @@
+from decimal import Decimal
+from typing import Annotated
+import datetime
+import logging
+
+import pydantic
+
+
+BM_UNKNOWN_SU = "BM UNKNOWN"
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def parse_date(v: str | datetime.date) -> datetime.date:
+    if isinstance(v, str):
+        return datetime.datetime.fromisoformat(v)
+    return v
+
+
+PositiveDecimalField = Annotated[Decimal, pydantic.Field(gt=0)]
+PositiveIntField = Annotated[int, pydantic.Field(gt=-1)]
+DateField = Annotated[datetime.datetime, pydantic.BeforeValidator(parse_date)]
+
+
+class BMNodeUsage(pydantic.BaseModel):
+    resource_class: Annotated[str, pydantic.Field(alias="Resource Class")]
+    project: Annotated[str, pydantic.Field(alias="Project")]
+    start_time: Annotated[DateField, pydantic.Field(alias="Start Time")]
+    expire_time: Annotated[DateField | None, pydantic.Field(alias="Expire Time")] = None
```

Comment:
I thought I commented it elsewhere, but it would be nice if we include the `resource` as well so we know exactly which node we billed for. Additionally, tracking the lease UUID will also be helpful. These may not be immediately useful when generating the billing/usage report, but it would be useuful information to have if we wanted to generate a report based on usage by each node.

Reply:
Running the script myself I realized it would also be useful for logging